src/calculator/SFA.DAS.ProviderPayments.Calc.PaymentsDue/PaymentsDueProcessor.cs

```diff
@@ -309,6 +313,89 @@ private void ApportionPaymentDuesOverPreviousPeriods(Provider provider, List<Req
                 AddPaymentsDue(provider, paymentsDue, earning, earning.EarnedValue, earning.CalendarMonth, earning.CalendarYear);
             }
         }
+
+        private void ApportionPaymentDuesOverPreviouslyPaidSameMonths(Provider provider, List<RequiredPayment> paymentsDue, PeriodEarning earning, RequiredPayment[] paymentHistory, decimal amountDue)
```

Comment:
This method looks like it could be refactored to make it easier to read/understand.  A good starting point might be wherever there is a comment to extract a method?